### ISSUE TITLE
ansible:  Missing dependencies

### ISSUE THREAD

**Issue by maxbrunet** [Homebrew/homebrew-core#34267]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

```yaml
HOMEBREW_VERSION: 1.8.2-96-g4021aa8
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: 4021aa80dfbaa369ee357a955c0216f9566133ac
Last commit: 32 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 50c4ae03d933a76b2671bfae6162be16e049ceb5
Core tap last commit: 32 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_DEV_CMD_RUN: 1
CPU: quad-core 64-bit broadwell
Homebrew Ruby: 2.3.7 => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Clang: 10.0 build 1000
Git: 2.19.1 => /usr/local/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 11.0.1
macOS: 10.14.1-x86_64
CLT: 10.1.0.0.1.1539992718
Xcode: N/A
```

---

To help us debug your issue please explain:

### What you were trying to do (and why)

Use [Ansible plugins](https://docs.ansible.com/ansible/2.5/plugins/plugins.html) (e.g. [`hashi_vault`](https://docs.ansible.com/ansible/2.5/plugins/lookup/hashi_vault.html) lookup which depends on [`hvac`](https://pypi.org/project/hvac/))

### What happened (include command output)

```shell
$ ansible localhost --module-name debug --args msg='{{ lookup("hashi_vault", "secret=path/to/secret") }}'
localhost | FAILED! => {
    "msg": "An unhandled exception occurred while running the lookup plugin 'hashi_vault'. Error was a <class 'ansible.errors.AnsibleError'>, original message: Please pip install hvac to use the hashi_vault lookup module."
}
```

### Step-by-step reproduction instructions (by running `brew install` commands)

* For example, Try to use the `hashi_vault` lookup plugin, it will fail on the missing `hvac` dependency:

```shell
$ ansible localhost --module-name debug --args msg='{{ lookup("hashi_vault", "secret=path/to/secret") }}'
localhost | FAILED! => {
    "msg": "An unhandled exception occurred while running the lookup plugin 'hashi_vault'. Error was a <class 'ansible.errors.AnsibleError'>, original message: Please pip install hvac to use the hashi_vault lookup module."
}
```

* Install `hvac` in the `virtualenv` provided by `brew` for `ansible`:

```shell
"$(brew --cellar ansible)/$(brew info ansible --json=v1 | jq -r '.[].linked_keg')/libexec/bin/pip" install hvac
```

* The first Ansible command should now work if you have a Vault instance properly configured:

```shell
$ ansible localhost --module-name debug --args msg='{{ lookup("hashi_vault", "secret=path/to/secret") }}'
localhost | SUCCESS => {
    "msg": {
        "mySecret": "This a secret"
    }
}
```

* Upgrade Ansible:

```shell
brew upgrade ansible
```

* Re-run the first command again, it should fail again because the new version of `ansible` uses a new `virtualenv`:

```shell
$ ansible localhost --module-name debug --args msg='{{ lookup("hashi_vault", "secret=path/to/secret") }}'
localhost | FAILED! => {
    "msg": "An unhandled exception occurred while running the lookup plugin 'hashi_vault'. Error was a <class 'ansible.errors.AnsibleError'>, original message: Please pip install hvac to use the hashi_vault lookup module."
}
```

### What you expected to happen

Dependencies are satisfied by the `brew` formula or at least preserved across upgrades.

---

### What has been discussed so far 

Such dependency has already been added for individual plugins such as [`consul_kv`](https://docs.ansible.com/ansible/latest/modules/consul_kv_module.html) (#20844) in the past.

I tried to cover them all once for all (#28839) (until next addition), but we agreed that this is not the best solution due to its maintainability and the fact that these dependencies would be bloat for most users.

A suggestion from @claco is to follow the upstream packaging practices such as the [Ansible PPA](https://launchpad.net/~ansible/+archive/ubuntu/ansible) and not include additional dependency by default, but allow their installation with `--with-*` option (these options would still need maintenance and are not recommended as @javian said).

Here is the the list of dependencies for the PPA:
https://github.com/ansible/ansible/blob/v2.7.2/packaging/debian/control#L11

Following upstream packaging practice would also mean removing the dependencies (e.g. `openstacksdk`, `python-consul`, `zabbix-api`, ...) which were already added for different plugins and modules.

Another possibility could be to improve the dependency management of the upstream, see
ansible/proposals#87 for example.

I'd like to also point out that Homebrew is not the recommended way to install Ansible. (ansible/ansible#10607)

To sum up the ideas:

* Add all the dependencies to the formula
* Follow upstream packaging practices and inject additional dependencies in some way
* Rework how dependencies are managed at the upstream level?
* Use `pip` instead of `brew` as documented by upstream

**Comment by MikeMcQuaid** [comment]
> but allow their installation with `--with-*` option (these options would still need maintenance and are not recommended as @javian said)

We are not interested in adding additional options to packages.



> * Use `pip` instead of `brew` as documented by upstream

If the upstream dependency versions are somehow pinned we could consider using `pip` within Homebrew to install dependencies?
